autoload/go/guru.vim

```diff
@@ -198,23 +197,26 @@ function! s:async_guru(args) abort
   endfunction
 
   let start_options = {
-        \ 'close_cb': function("s:close_cb"),
+        \ 'close_cb': funcref("s:close_cb"),
+        \ 'in_mode': 'raw',
+        \ 'out_mode': 'nl',
+        \ 'err_mode': 'nl',
         \ }
 
-  if &modified
-    let l:tmpname = tempname()
-    call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
-  endif
-
```

Comment:
I realize now that this code wasn't working quite the way I thought it was; it was, in fact, honoring the -modified flag as guru_cmd constructed it. I still think there's value in writing directly to the channel instead of writing to a temp file first and reading stdin from the file, though.